dhis_android_analytics/src/main/java/dhis2/org/analytics/charts/data/Graph.kt

```diff
@@ -24,13 +24,32 @@ data class Graph(
     val errorMessage: String? = null,
 ) {
 
-    private fun minDate() = series.minOfOrNull { serie ->
-        serie.coordinates.minOf { point -> point.eventDate }
-    }?.toInstant()?.atZone(ZoneId.systemDefault())?.toLocalDate() ?: LocalDate.now()
+    private fun minDate(): LocalDate? {
+        val coordinatesNotEmpty = series.minOfOrNull { serie ->
+            serie.coordinates.isNotEmpty()
+        }
+        return if (series.isNotEmpty() && coordinatesNotEmpty == true) {
+            val minDate = series.minOfOrNull { serie ->
+                serie.coordinates.minOf { point -> point.eventDate }
+            }?.toInstant()?.atZone(ZoneId.systemDefault())?.toLocalDate() ?: LocalDate.now()
+
+            return minDate
+        } else { LocalDate.now() }
+    }
 
-    private fun maxDate() = series.maxOfOrNull { serie ->
-        serie.coordinates.maxOf { point -> point.eventDate }
-    }?.toInstant()?.atZone(ZoneId.systemDefault())?.toLocalDate() ?: LocalDate.now()
+    private fun maxDate(): LocalDate? {
+        val coordinatesNotEmpty = series.maxOfOrNull { serie ->
```

Comment:
Couldn't using maxOfOrNull to determine a boolean give us wrong data?

Reply:
The problem is that the check is done over a list of SerieDate that contains a list of Coordinates. I couldn't simply add minOfOrNull to the nested check as it formats the value later. otherwise it will always return Local date so I believe correct output will be maintained with the new changes. it will return localDate if coordinate list is empty.